presto-verifier/src/main/java/com/facebook/presto/verifier/framework/InsertExtendedVerification.java

```diff
@@ -0,0 +1,177 @@
+                snapshotQueryConsumer,
+                snapshotQueries);
+    }
+
+    @Override
+    public DataMatchResult verify(
+            QueryObjectBundle control,
+            QueryObjectBundle test,
+            Optional<QueryResult<Void>> controlQueryResult,
+            Optional<QueryResult<Void>> testQueryResult,
+            ChecksumQueryContext controlChecksumQueryContext,
+            ChecksumQueryContext testChecksumQueryContext)
+    {
+        // 0. Data verification
+        DataMatchResult dataMatchResult = super.verify(control, test, controlQueryResult, testQueryResult, controlChecksumQueryContext, testChecksumQueryContext);
+        if (!dataMatchResult.isMatched()) {
+            return dataMatchResult;
+        }
+
+        // 1. Partition verification
+        List<Column> controlPartitionColumns = null;
+        List<Column> testPartitionColumns = null;
+        try {
+            controlPartitionColumns = getColumns(getHelperAction(), typeManager, formPartitionTableName(control.getObjectName()));
+            testPartitionColumns = getColumns(getHelperAction(), typeManager, formPartitionTableName(test.getObjectName()));
+        }
+        catch (Exception e) {
+            // For un-partitioned tables, this query would fail. Directly return data checksum mach result.
+            return dataMatchResult;
+        }
+
+        List<Column> controlDataColumns = getDataColumn(getColumns(getHelperAction(), typeManager, control.getObjectName()), ImmutableSet.copyOf(controlPartitionColumns));
+        List<Column> testDataColumns = getDataColumn(getColumns(getHelperAction(), typeManager, test.getObjectName()), ImmutableSet.copyOf(testPartitionColumns));
+        List<ChecksumResult> controlPartitionChecksum = runPartitionChecksum(control, controlPartitionColumns, controlDataColumns, controlChecksumQueryContext, CONTROL_PARTITION_CHECKSUM);
+        List<ChecksumResult> testPartitionChecksum = runPartitionChecksum(test, testPartitionColumns, testDataColumns, testChecksumQueryContext, TEST_PARTITION_CHECKSUM);
+        if (controlPartitionChecksum.size() != testPartitionChecksum.size()) {
+            return new DataMatchResult(
+                    PARTITION_DATA,
+                    ROW_COUNT_MISMATCH,
+                    Optional.empty(),
+                    OptionalLong.of(controlPartitionChecksum.size()),
+                    OptionalLong.of(testPartitionChecksum.size()),
+                    ImmutableList.of());
+        }
+        for (int i = 0; i < controlPartitionChecksum.size(); i++) {
+            DataMatchResult partitionMatchResult = match(
+                PARTITION_DATA,
+                checksumValidator,
+                controlDataColumns,
+                testDataColumns,
+                    controlPartitionChecksum.get(i),
+                    testPartitionChecksum.get(i));
+            if (!partitionMatchResult.isMatched()) {
+                return partitionMatchResult;
+            }
```

Comment:
It looks you don't include the particular partition that has the mismatch, do you? If so, it might not be quite useful for users to debug.

Can you merge the DataMatchResult and show all that partitions that have mismatch? That'll be useful for debugging.